### PR TITLE
DomainTip: Correct and simplify rendering logic

### PR DESCRIPTION
**Pull request by jsnmoon**
This refactors the `<DomainTip />` component to show correct promotional notices. Currently, this component is used only at [`/stats/insights/`](https://wordpress.com/stats/insights/).

- Renders nothing if:
  - the site object or the `siteSlug` value are missing.
  - it is a Jetpack site.

- Promotes plan upgrades if the site is on the free plan and is part of the Domains With Plans Only (DWPO) initiative. 

plan_nudge

- Promotes claiming a domain name if the site is on a paid plan and has a domain credit.

domain_credit

- Promotes buying a domain name in all other cases. This should only apply to two scenarios:
  1. The site is on a paid plan without domain credit
  2. The site is exempt from the Domains With Plans Only initiative. 

purchase_domain

# Testing Instructions

For testers included in the DWPO initiative:

1. Spin up this branch locally.
2. Navigate to [/stats/insights/](http://calypso.localhost:3000/stats/insights/) and select a site on a free plan. 
3. Ensure that you can see a plan upgrade nudge.
4. Repeat steps 2 and 3 for a site on a paid plan with a domain credit. Ensure that the nudge mentions claiming a free custom domain.
5. Repeat steps 2 and 3 for a site on a paid plan without a domain credit. Ensure that the nudge mentions purchasing a custom domain.

For testers exempt from the DWPO initiative:

1. Spin up this branch locally.
2. Navigate to [/stats/insights/](http://calypso.localhost:3000/stats/insights/) and select a site on a free plan. 
3. Ensure that you see a custom domain purchase nudge.
4. This is the same as step 4 above.
5. This is the same as step 5 above.

### CODE REVIEW

**Comment by klimeryk** [comment]
Thanks for the improvements! I've tested all the cases I could think of and it works as expected... except one: what about the case when the site is on a paid plan _and_ has used the domain credit (for example, to get a domain mapping or registration). On production, we do _not_ show the notice, but with your PR we do. I wonder if this is intentional - I mean, someone _could_ purchase a domain through it, but I guess if they have a plan _and_ a domain name, we should not try to upsell them to buy another one 🤔

**Comment by jsnmoon** [comment]
> what about the case when the site is on a paid plan and has used the domain credit (for example, to get a domain mapping or registration). On production, we do not show the notice, but with your PR we do. I wonder if this is intentional - I mean, someone could purchase a domain through it, but I guess if they have a plan and a domain name, we should not try to upsell them to buy another one 🤔 (@klimeryk)

I intentionally added that handling, but I did so without product context that you probably have. I'll tweak the logic so that we no longer suggest custom domain purchases for sites that have paid plans and no domain credit.

**Comment by jsnmoon** [comment]
@klimeryk: Fixed! Should be ready for another 👀

File: client/my-sites/domain-tip/index.jsx [hunk 1]
```diff
@@ -38,79 +37,67 @@ class DomainTip extends React.Component {
...
-				feature={ FEATURE_CUSTOM_DOMAIN }
-				event={ event }
-			/>
+			<div className={ classNames( 'domain-tip', this.props.className ) }>
+				<UpgradeNudge
+					event={ `domain_tip_${ this.props.event }` }
+					feature={ FEATURE_CUSTOM_DOMAIN }
+					shouldDisplay
+					message={ this.props.translate( 'Upgrade your plan to register a custom domain.' ) }
+					title={ this.props.translate( 'Get a custom domain' ) }
```
Comment (a8ci18n) [inline client/my-sites/domain-tip/index.jsx]: Hi! I've found a possible matching string that has already been translated **33** times:
`translate( 'Get a free Custom Domain' )` *ES Score: 12*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/fix/domain-tip-render-logic)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

Comment (jsnmoon) [reply client/my-sites/domain-tip/index.jsx]: 🔴

**Pull request merged**